var/spack/repos/builtin/packages/camp/package.py

```diff
@@ -110,9 +110,9 @@ def cmake_args(self):
             options.append("-DHIP_ROOT_DIR={0}".format(spec["hip"].prefix))
 
             archs = self.spec.variants["amdgpu_target"].value
-            options.append("-DCMAKE_HIP_ARCHITECTURES={0}".format(archs))
-            options.append("-DGPU_TARGETS={0}".format(archs))
-            options.append("-DAMDGPU_TARGETS={0}".format(archs))
+            options.append("-DCMAKE_HIP_ARCHITECTURES={0}".format(archs[0]))
```

Comment:
You should pass all architectures, like this:
```suggestion
            options.append(self.define("CMAKE_HIP_ARCHITECTURES", archs))
```
See https://github.com/spack/spack/blob/a14f10e8825fd25e0615095feaa548b1938fffac/lib/spack/spack/build_systems/cmake.py#L519

Reply:
Thanks for your comment! You are right that all architectures should be passed. The problem I was trying to fix was that the architectures were getting passed as, for example, `"-DCMAKE_HIP_ARCHITECTURES=('gfx90a',)"` resulting in a syntax error due to the parentheses/formatting. I think my updated changes fix this better by turning the tuple of architectures into a semi-colon separated list.